backend/corpora/lambdas/api/v1/common.py

```diff
@@ -35,3 +37,32 @@ def delete_dataset_common(db_session: Session, dataset: Dataset, token_info: dic
                 original.create_revision(dataset.collection.id)  # Restore the original dataset and S3 assets
             dataset.asset_deletion()  # Delete the S3 assets and database rows.
             dataset.delete()  # Delete the dataset row.
+
+
+def reshape_for_curation_api_and_is_allowed(collection, token_info, allow_access=False):
```

Comment:
an existing reshape_for_api exists as a Collections class method. Should this also be a Collections class method for consistency? Perhaps with a rename for clarity distinguishing the use case difference

Reply:
I'd like to retain the reusability of this function while also avoiding the messiness of having a function that is both an in-place modifier and also returns a bool evaluating auth, while maintaining a flag param to essentially ignore the auth evaluation. 

what if we separated the access_type evaluation into one function (returning either "WRITE", "READ", or None) and then handled the in-place collection reshape in a separate `reshape_for_curation_api`? In the non-UUID endpoint, we can skip over collections with access type None and in the UUID endpoint we can call the reshape anyway and pass in the access_type as a param to avoid rework. Thoughts?

Reply:
Move this to the function handler that uses it. This function is only used in one place, and doesn't seem like something that will be resued.

>an existing reshape_for_api exists as a Collections method. Should this also be a Collections method for consistency? Perhaps with a rename for clarity distinguishing the use case difference

We should move the `Collection.reshape_for_api` out of the `Collection` class, but that can be done later. We want to move away from adding more functionality to the classes that inherit `Entity` and keep the code specific to the APIs closer to the handlers.

Reply:
It's used in two places, in the handlers for both `GET /collections` and `GET /collections/{collection_uuid}`